functions/Set-DbaSpConfigure.ps1

```diff
@@ -52,78 +55,91 @@ Set-DbaSpConfigure -SqlServer localhost -configs XPCmdShellEnabled -value 1 -Wha
+		{
+			Stop-Function -Message "You must select one or more configurations to modify"
+		}
 	}
 	
-	PROCESS
+	process
 	{
-		FOREACH ($instance in $SqlServer)
+		if (Test-FunctionInterrupt) { return }
+		
+		foreach ($instance in $SqlInstance)
 		{
-			TRY
+			try
 			{
 				$server = Connect-SqlServer -SqlServer $instance -SqlCredential $sqlcredential
 			}
-			CATCH
+			catch
 			{
-				Write-Warning "Failed to connect to: $instance"
-				continue
+				Stop-Function -Message "Failed to connect to: $instance" -Continue
 			}
 			
-            #Grab the current config value
-            $currentValues = ($server.Configuration.$configs)
-            $currentRunValue = $currentValues.RunValue
-            $minValue = $currentValues.Minimum
-            $maxValue = $currentValues.Maximum
-            $isDynamic = $currentValues.IsDynamic
-            
-            #Let us not waste energy setting the value to itself
-            if ($currentRunValue -eq $value) { write-warning "Value to set is the same as the existing value. No work being performed." ;break; }
-
-            #Going outside the min/max boundary can be done, but it can break SQL, so I don't think allowing that is wise at this juncture
-            if ($value -le $minValue -or $value -gt $maxValue) { Write-Error "Value out of range for $($configs) (min: $($minValue) - max $($maxValue))"; break; }
-
-            
-            If ($Pscmdlet.ShouldProcess($SqlServer, "Adjusting server configuration $($configs) from $($currentValue) to $($value)."))
-            {
-                try
-                {
-                    $server.Configuration.$configs.ConfigValue = $value;
-                    $server.Configuration.Alter();
-
-                    #If it's a dynamic setting we're all clear, otherwise let the user know that SQL needs to be restarted for the change to take
-                    if ($isDynamic -eq $true)
-                        {
-                            Write-Output "Config for $($configs) changed. Old value: $($currentRunValue)  New Value: $($value)"
-                        }
-                    else
-                        {
-                            Write-Warning "Config set for $($configs), but restart of SQL Server is required for the new value ($($value)) to be used (old value: $($value))"
-                        }
-
-                }
-                catch
-                {
-                    Write-Error "Unable to change config setting - $($Error)"
-                }
-            }
+			#Grab the current config value
+			$currentValues = ($server.Configuration.$configs)
+			$currentRunValue = $currentValues.RunValue
+			$minValue = $currentValues.Minimum
+			$maxValue = $currentValues.Maximum
+			$isDynamic = $currentValues.IsDynamic
 			
+			#Let us not waste energy setting the value to itself
+			if ($currentRunValue -eq $value)
+			{
+				Stop-Function -Message "Value to set is the same as the existing value. No work being performed."
```

Comment:
Should have the -Continue parameter. Input may be good for next instance.
Add Parameter: '-Target $Instance', making troubleshooting easier.